src/embedding.rs

```diff
@@ -77,6 +83,46 @@ impl MatrixWrapper for TwoDimVectorMatrix {
         }
     }
 
+    fn init_with_eigenvectors(
+        rows: usize,
+        cols: usize,
+        sparse_matrix_reader: Arc<SparseMatrix>,
+        order: TruncatedOrder,
+    ) -> Self {
+        let mut discrete_laplacian = Array2::zeros((rows, rows));
```

Comment:
This is probably not feasible. Cleora is all about using a sparse matrix representation. Its a core assumption in cleora. 

And this assumption is broken here - we have dense [row x row] matrix. So why bother with sparse `sparse_matrix_reader.entries` in the first place if we would have dense allocation here anyway?

This laplacian algo should probably be sparsely calculated without allocating full matrix